src/array.jl

```diff
@@ -552,18 +552,66 @@ function copyto!(dest::CatArrOrSub{T, N, R}, dstart::Integer,
     dest
 end
 
-copyto!(dest::CatArrOrSub, src::CatArrOrSub) =
+function copyto!(dest::CatArrOrSub{T1, N, R}, dstart::Integer,
+                 src::AbstractArray{T2,N}, sstart::Integer,
+                 n::Integer) where {T1, T2, N, R}
+    n == 0 && return dest
+    n < 0 && throw(ArgumentError(string("tried to copy n=", n, " elements, but n should be nonnegative")))
```

Comment:
why not this:
```suggestion
    n < 0 && throw(ArgumentError(string("tried to copy n=$n elements, but n should be nonnegative")))
```

Reply:
I think I just copied this from Base, where it may have be written before string interpolation even existed. Fixed. :-)